### DISCUSSION TITLE
Multiple loaders per route?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2614] [Q&A] [answered]
Not sure if this is something I will actually need to do, but the use case is as follows:
- I have a route which fetches data from a slow API and a fast API
- I want to start rendering the route with the data from the fast API as soon as possible, if the slow API data comes in later that's ok

This seems like it would require multiple loaders for a single route. Is this possible?

Example: I'm displaying a page that has comments. The comments are non-essential information that come from a slower API than the fast API that directly serves the page contents.

**Comment by sergiodxa** [comment, marked as answer]
What you need is to:
1. Try to make the slow api faster
2. Cache the slow api if you can’t make it faster
3. Move the fetch to the slow api to the browser using a resource route and useFetcher instead of doing it on the route loader if you can’t cache it or make it faster

**Reply by juniorforlife** [reply]
@sergiodxa Why is moving the slow api to the resource route & useFetcher is better than doing it on the route loader?

**Reply by sergiodxa** [reply]
You can send the initial response faster, then fetch the slow part client side.

The answer is also old, you can now use `defer` to start sending a response faster while slow data is still being fetched, but allowing you to start the slow fetch earlier.

**Reply by juniorforlife** [reply]
thank you. I'll try that

**Comment by ariofrio** [comment]
I think you can just use `defer` for the slow data: https://remix.run/docs/en/1.15.0/guides/streaming

It allows you to to have the loader return partial data at multiple times.

**Reply by juniorforlife** [reply]
thank you. I'll try that
